lib/Doctrine/DBAL/Driver/PDOConnection.php

```diff
@@ -120,7 +125,19 @@ public function quote($input, $type = \PDO::PARAM_STR)
      */
     public function lastInsertId($name = null)
     {
-        return parent::lastInsertId($name);
+        if (null !== $name) {
+            return parent::lastInsertId($name);
+        }
+
+        $lastInsertId = parent::lastInsertId($name);
+
+        // The last insert ID is reset to "0" in certain situations by some implementations,
+        // therefore we keep the previously set insert ID locally.
+        if ($lastInsertId) {
```

Comment:
Yes, sequences are indeed not affected, but we still need to consider scenarios where something like an `AUTO_INCREMENT` started at `-10`, and then at some point reaches `0`. We can't handle that, but we have to at least specify that to users (we only support positive insert identifiers, starting from `1`, while all, other scenarios lead to edge cases)